src/lwc/signals/core.ts

```diff
@@ -103,31 +122,36 @@ function computedGetter<T>(node: ComputedNode<T>) {
  * ```
  *
  * @param fn The function that returns the computed value.
+ * @param props Options to configure the computed value.
  */
-function $computed<T>(fn: ComputedFunction<T>): ReadOnlySignal<T> {
-  const computedNode: ComputedNode<T> = {
-    signal: $signal<T | undefined>(undefined),
-    error: null,
-    state: UNSET
-  };
-
-  $effect(() => {
-    if (computedNode.state === COMPUTING) {
-      throw new Error("Circular dependency detected");
-    }
-
-    try {
-      computedNode.state = COMPUTING;
-      computedNode.signal.value = fn();
-      computedNode.error = null;
-      computedNode.state = READY;
-    } catch (error) {
-      computedNode.state = ERRORED;
-      computedNode.error = error;
-    }
+function $computed<T>(
+  fn: ComputedFunction<T>,
+  props?: Partial<ComputedProps<T>>
+): ReadOnlySignal<T> {
+  const computedSignal: Signal<T | undefined> = $signal(undefined, {
+    track: true
   });
-
-  return computedGetter(computedNode);
+  $effect(
+    () => {
+      if (props?.errorHandler) {
+        // If this computed has a custom errorHandler, then error
+        // handling occurs in the computed function itself.
+        try {
+          computedSignal.value = fn();
+        } catch (error) {
+          computedSignal.value = props.errorHandler(error);
+        }
+      } else {
+        // Otherwise, the error handling is done in the $effect
+        computedSignal.value = fn();
```

Comment:
can the props be passed backed to the function here, to have access to identifier?